freeipa/src/main/java/com/sequenceiq/freeipa/cost/FreeIpaCostService.java

```diff
@@ -0,0 +1,96 @@
+package com.sequenceiq.freeipa.cost;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+import com.sequenceiq.cloudbreak.auth.altus.EntitlementService;
+import com.sequenceiq.cloudbreak.common.cost.RealTimeCost;
+import com.sequenceiq.cloudbreak.cost.CostCalculationNotEnabledException;
+import com.sequenceiq.cloudbreak.cost.co2.CarbonCalculatorService;
+import com.sequenceiq.cloudbreak.cost.model.ClusterCostDto;
+import com.sequenceiq.cloudbreak.cost.usd.UsdCalculatorService;
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.service.stack.StackService;
+
+@Service
+public class FreeIpaCostService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FreeIpaCostService.class);
+
+    @Inject
+    private CarbonCalculatorService carbonCalculatorService;
+
+    @Inject
+    private UsdCalculatorService usdCalculatorService;
+
+    @Inject
+    private FreeIpaInstanceTypeCollectorService instanceTypeCollectorService;
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private EntitlementService entitlementService;
+
+    private boolean usdCalculationEnabled;
+
+    private boolean co2CalculationEnabled;
+
+    public Map<String, RealTimeCost> getCosts(List<String> environmentCrns) {
+        checkIfCostCalculationIsEnabled();
+        errorIfNeitherCostCalculationFeaturesAreEnabled();
+        Map<String, RealTimeCost> realTimeCosts = new HashMap<>();
+        String accountId = ThreadBasedUserCrnProvider.getAccountId();
+
+        for (String envCrn : environmentCrns) {
+            Stack stack = stackService.getByEnvironmentCrnAndAccountIdWithLists(envCrn, accountId);
```

Comment:
same here, for freeipa there is a method in StackService called `getMultipleByEnvironmentCrnOrChildEnvironmantCrnAndAccountId`